### PR TITLE
Fix wrong error message

### LINKED ISSUE #768: Wrong error when card is invalid
**Describe the bug**

image

This screen is displayed when the card is invalid:

```
[JettyServerThreadPool-19] INFO io.javalin.Javalin - JAVALIN REQUEST DEBUG LOG:
Request: POST [/]
    Matching endpoint-handlers: [BEFORE=*, POST=/, AFTER=*]
    Headers: {Accept=*/*, User-Agent=Dart/2.18 (dart:io), Host=localhost:8000, Accept-Encoding=gzip, Content-Length=350, Content-Type=application/json}
    Cookies: {}
    Body: {"operationName":null,"variables":{"project":"nuernberg.sozialpass.app","card":{"cardInfoHashBase64":"x80WFKWNZy0iCv0Q04ylqM+twCvTZEdxP4SuVbgcJT8=","codeType":"dynamic","totp":505474}},"query":"query CardVerificationByHash($project: String!, $card: CardVerificationModelInput!) {\n  cardValid: verifyCardInProject(project: $project, card: $card)\n}"}
    QueryString: null
    QueryParams: {}
    FormParams: {}
Response: [200 OK], execution took 183.60 ms
    Headers: {Date=Mon, 30 Jan 2023 09:28:13 GMT, Content-Type=application/json}
    Body is 28 bytes (starts on next line):
    {"data":{"cardValid":false}}
```

**Expected behavior**
We should display that the card is not valid.

### PR DESCRIPTION
**Pull request by michael-markl**
Closes #768 .

I could not reproduce the original error. I suspect, it was actually a connection failed error, which we didn't handle correctly

### CODE REVIEW

File: .idea/runConfigurations/Run__env_local_buildConfig_bayern_.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <component name="ProjectRunConfigurationManager">
   <configuration default="false" name="Run (env:local+buildConfig:bayern)" type="FlutterRunConfigurationType" factoryName="Flutter">
     <option name="additionalArgs" value="--dart-define=environment=local --flavor Bayern" />
-    <option name="attachArgs" value="--dart-define=environment=local --flavor Bayern" />
```
Comment (maxammann) [inline .idea/runConfigurations/Run__env_local_buildConfig_bayern_.xml:4]: Why did you remove the flavors here? The flavor is required for selecting the proper app name for instance.

File: frontend/lib/identification/connection_failed_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+class ConnectionFailedDialog extends StatelessWidget {
+  final String reason;
+
+  const ConnectionFailedDialog({super.key, required this.reason});
+
+  @override
+  Widget build(BuildContext context) {
+    return InfoDialog(
+      title: "Keine Verbindung möglich",
```
Comment (maxammann) [inline frontend/lib/identification/connection_failed_dialog.dart:12]: Maybe add an explanation here:

```suggestion
      title: "Keine Verbindung möglich. Dies bedeutet nicht dass der QR Code ungültig ist. Versuchen Sie erneut die Karte zu scannen.",
```

**Review by maxammann — APPROVED** [review]
Except for the build configs, looks good!

File: .idea/runConfigurations/Run__env_local_buildConfig_bayern_.xml (hunk 1 above)
Comment (michael-markl) [reply .idea/runConfigurations/Run__env_local_buildConfig_bayern_.xml:4]: Flutter attach does not work with flavors (ie. the flavor option is not accepted). Luckily, the attach args can be specified separately (I only adjusted the attach args here).

File: frontend/lib/identification/connection_failed_dialog.dart (hunk 2 above)
Comment (michael-markl) [reply frontend/lib/identification/connection_failed_dialog.dart:12]: There's an explanation in Text(reason) below.

File: .idea/runConfigurations/Run__env_local_buildConfig_bayern_.xml (hunk 1 above)
Comment (maxammann) [reply .idea/runConfigurations/Run__env_local_buildConfig_bayern_.xml:4]: Ah its the attach! Sorry!
